src/dali_backend.cc

```diff
@@ -180,33 +185,26 @@ class DaliModelInstance : public ::triton::backend::BackendModelInstance {
   void Execute(const std::vector<TritonRequest>& requests) {
     DeviceGuard dg(GetDaliDeviceId());
     int total_batch_size = 0;
-    TimeInterval batch_compute_interval{};
-    TimeInterval batch_exec_interval{};
-    start_timer_ns(batch_exec_interval);
-    for (size_t i = 0; i < requests.size(); i++) {
-      TimeInterval req_exec_interval{};
-      start_timer_ns(req_exec_interval);
-      auto response = TritonResponse::New(requests[i]);
-      RequestMeta request_meta;
-      TritonError error{};
-      try {
-        request_meta = ProcessRequest(response, requests[i]);
-      } catch (...) { error = ErrorHandler(); }
-
-      if (i == 0) {
-        batch_compute_interval.start = request_meta.compute_interval.start;
-      } else if (i == requests.size() - 1) {
-        batch_compute_interval.end = request_meta.compute_interval.end;
-      }
-
-      end_timer_ns(req_exec_interval);
-      ReportStats(requests[i], req_exec_interval, request_meta.compute_interval, !error);
-      SendResponse(std::move(response), std::move(error));
-
-      total_batch_size += request_meta.batch_size;
+    TimeInterval exec_interval{};
+    start_timer_ns(exec_interval);
+    std::vector<TritonResponse> responses;
+    responses.reserve(requests.size());
+    for (auto& request : requests) {
```

Comment:
To match the ProcessRequests (with the GenerateInputs/AllocateOutputs) and SendResposnes I would suggest some AllocateRepsonses or something similar.

Reply:
done